src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -0,0 +1,682 @@
+                    }
+                }
+            }
+        }
+
+        public bool Exists
+        {
+            get
+            {
+                unsafe
+                {
+                    return _windowPtr != null && !Glfw.WindowShouldClose(_windowPtr);
+                }
+            }
+        }
+        
+        public unsafe WindowState WindowState
+        {
+            get
+            {
+                if (Glfw.GetWindowAttrib(_windowPtr, (int)WindowHint.Iconified) == 1)
+                {
+                    return WindowState.Minimized;
+                }
+
+                if (Glfw.GetWindowMonitor(_windowPtr) != null)
+                {
+                    return WindowState.Fullscreen;
+                }
+
+                var mode = Glfw.GetVideoMode(CurrentMonitor);
+
+                Glfw.GetWindowSize(_windowPtr, out var windowWidth, out var windowHeight);
+
+                if (mode->width == windowWidth && mode->height == windowHeight)
+                {
+                    return WindowState.Maximized;
+                }
+
+                return WindowState.Normal;
+            }
+            set
+            {
+                switch (value)
+                {
+                    case WindowState.Normal:
+                        Glfw.RestoreWindow(_windowPtr);
+                        break;
+                    case WindowState.Minimized:
+                        Glfw.IconifyWindow(_windowPtr);
+                        break;
+                    case WindowState.Maximized:
+                        Glfw.MaximizeWindow(_windowPtr);
+                        break;
+                    case WindowState.Fullscreen:
+                        var mode = Glfw.GetVideoMode(CurrentMonitor);
+                        Glfw.SetWindowMonitor(_windowPtr, CurrentMonitor, 0, 0, mode->width, mode->height, mode->refreshRate);
+                        break;
+                }
+            }
+        }
+
+        private WindowBorder _windowBorder;
+
+        public WindowBorder WindowBorder
+        {
+            get
+            {
+                unsafe
+                {
+                    if (Glfw.GetWindowAttrib(_windowPtr, (int)WindowHint.Decorated) == 0)
+                    {
+                        return WindowBorder.Hidden;
+                    }
+                    
+                    return Glfw.GetWindowAttrib(_windowPtr, (int)WindowHint.Resizable) == 0
+                        ? WindowBorder.Fixed : WindowBorder.Resizable;
+                }
+            }
+            
+            //If GLFW 3.3 is supported by the time OpenTK 4.0 is ready, this will be implemented,
+            //but we want to avoid making features that depend on development versions of libraries.
+            set => throw new NotImplementedException("Cannot be implemented in GLFW 3.2.");
+        }
+
+        public Box2 Bounds
+        {
+            get => new Box2(Location, Size);
```

Comment:
here you pass Location and Size as parameters, whereas the Box2 ctor does not take size as second parameter but, the position of the bottom-right corner....
Should use Box2.FromDimensions...

Reply:
Crap, sorry. Fixed.